src/components/survey/other-structure.tsx

```diff
@@ -0,0 +1,308 @@
+  const [ creatingOtherStructure, setCreatingOtherStructure ] = React.useState(false);
+  const [ deletingOtherStructureId, setDeletingOtherStructureId ] = React.useState<number>();
+  const [ newOtherStructureName, setNewOtherStructureName ] = React.useState("");
+  const [ newOtherStructureDesc, setNewOtherStructureDesc ] = React.useState("");
+
+  /**
+   * Fetch building
+   */
+  const fetchBuilding = async () => {
+    if (!keycloak?.token || !surveyId) {
+      return;
+    }
+
+    try {
+      const fetchedBuildings = await Api.getBuildingsApi(keycloak.token).listBuildings({
+        surveyId: surveyId
+      });
+
+      if (fetchedBuildings.length !== 1) {
+        return;
+      }
+
+      setBuilding(fetchedBuildings[0]);
+    } catch (error) {
+      errorContext.setError(strings.errorHandling.buildings.list, error);
+    }
+  };
+
+  React.useEffect(() => {
+    fetchBuilding();
+  }, []);
+
+  /**
+   * Updates building
+   *
+   * @param updatedOwnerInformation updated owner information
+   */
+  const updateBuilding = async (updatedOwnerInformation: Building) => {
+    if (!keycloak?.token || !building?.id) {
+      return;
+    }
+
+    try {
+      Api.getBuildingsApi(keycloak.token).updateBuilding({
+        surveyId: surveyId,
+        buildingId: building.id,
+        building: updatedOwnerInformation
+      });
+    } catch (error) {
+      errorContext.setError(strings.errorHandling.buildings.update, error);
+    }
+  };
+
+  /**
+   * Delete other structure confirm handler
+   */
+  const onDeleteOtherStructureConfirm = async () => {
+    if (!building?.otherStructures || deletingOtherStructureId === undefined) {
+      return;
+    }
+
+    const updatedOtherStructures = building.otherStructures;
+
+    updatedOtherStructures.splice(deletingOtherStructureId, 1);
+
+    const updatedBuilding: Building = {
+      ...building,
+      otherStructures: updatedOtherStructures
+    };
+    await updateBuilding(updatedBuilding);
+
+    setDeletingOtherStructureId(undefined);
+  };
+
+  /**
+   * Create other structure confirm handler
+   */
+  const onCreateOtherStructureConfirm = async () => {
+    if (!building?.otherStructures) {
+      return;
+    }
+
+    const newOtherStructure: OtherStructure = {
+      name: newOtherStructureName,
+      description: newOtherStructureDesc
+    };
+
+    const updatedBuilding: Building = {
+      ...building,
+      otherStructures: [ ...building.otherStructures, newOtherStructure ]
+    };
+    await updateBuilding(updatedBuilding);
+
+    setNewOtherStructureName("");
+    setNewOtherStructureDesc("");
+    setCreatingOtherStructure(false);
+    fetchBuilding();
+  };
+
+  /**
+   * Event Handler set building other structure prop
+   * 
+   * @param index index
+   */
+  const onBuildingOtherStructurePropChange: (index: number) => React.ChangeEventHandler<HTMLTextAreaElement | HTMLInputElement> = (index: number) => ({
+    target
+  }) => {
```

Comment:
This can be simplified to

  const onBuildingOtherStructurePropChange = (index: number) => ({ target }) => {

Reply:
Compiler is going to complain target has any type

Reply:
```suggestion
  /**
   * Event Handler set building other structure prop
   * 
   * @param index index
   */
const onBuildingOtherStructurePropChange = (index: number): React.ChangeEventHandler<HTMLTextAreaElement | HTMLInputElement> => ({ target }) => {
```